File: Sorting/insertion_sort.py
lst = []
size = int(input("\nEnter size of the list: \t"))
for i in range(size):
    elements = eval((input("Enter the element: \t")))
    lst.append(elements)
for i in range(1, len(lst)):
        key = lst[i]
        j = i - 1
        while j >= 0 and key < lst[j]:
            lst[j + 1] = lst[j]
            j -= 1
        lst[j + 1] = key
print('\nThe sorted list: \t', lst)
print('\n')

File: Sorting/Insertion_Sort.cpp
#include <iostream>
using namespace std;
int main()
{
    int i,j,n,key, arr[30];
    cout<<"Enter the size of array: ";
    cin>>n;
    cout<<"\nEnter the elements\n";
    for(i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    for(i=1;i<=n-1;i++)
    {
       key=arr[i];
       j=i-1;
       while((key<arr[j])&&(j>=0))
         {
            arr[j+1]=arr[j];
            j=j-1;
         }
     arr[j+1]=key;
    }
    cout<<"Array after insertion sort: ";
    for(i=0;i<n;i++)
    {
        cout<<arr[i]<<" ";
    }
    return 0;
}
